server/users_test.go

```diff
@@ -31,900 +30,356 @@ func TestService_NewSourceUser(t *testing.T) {
-		}
-		if tt.wantContentType != "" && content != tt.wantContentType {
-			t.Errorf("%q. RemoveSourceUser() = %v, want %v", tt.name, content, tt.wantContentType)
-		}
-		if tt.wantBody != "" && string(body) != tt.wantBody {
-			t.Errorf("%q. RemoveSourceUser() = \n***%v***\n,\nwant\n***%v***", tt.name, string(body), tt.wantBody)
-		}
+			s.RemoveUser(tt.args.w, tt.args.r)
+
+			resp := tt.args.w.Result()
+
+			if resp.StatusCode != tt.wantStatus {
+				t.Errorf("%q. RemoveUser() = %v, want %v", tt.name, resp.StatusCode, tt.wantStatus)
+			}
+		})
 	}
 }
 
-func TestService_UpdateSourceUser(t *testing.T) {
+func TestService_UpdateUser(t *testing.T) {
 	type fields struct {
-		SourcesStore chronograf.SourcesStore
-		TimeSeries   server.TimeSeriesClient
-		Logger       chronograf.Logger
-		UseAuth      bool
+		UsersStore chronograf.UsersStore
+		Logger     chronograf.Logger
 	}
 	type args struct {
-		w *httptest.ResponseRecorder
-		r *http.Request
+		w    *httptest.ResponseRecorder
+		r    *http.Request
+		user *chronograf.User
 	}
 	tests := []struct {
 		name            string
 		fields          fields
 		args            args
-		ID              string
-		UID             string
+		id              string
 		wantStatus      int
 		wantContentType string
 		wantBody        string
 	}{
 		{
-			name: "Update user password for data source",
-			args: args{
-				w: httptest.NewRecorder(),
-				r: httptest.NewRequest(
-					"POST",
-					"http://server.local/chronograf/v1/sources/1",
-					ioutil.NopCloser(
-						bytes.NewReader([]byte(`{"name": "marty", "password": "the_lake"}`)))),
-			},
+			name: "Update a Chronograf user",
 			fields: fields{
-				UseAuth: true,
-				Logger:  log.New(log.DebugLevel),
-				SourcesStore: &mocks.SourcesStore{
-					GetF: func(ctx context.Context, ID int) (chronograf.Source, error) {
-						return chronograf.Source{
-							ID:       1,
-							Name:     "muh source",
-							Username: "name",
-							Password: "hunter2",
-							URL:      "http://localhost:8086",
-						}, nil
-					},
-				},
-				TimeSeries: &mocks.TimeSeries{
-					ConnectF: func(ctx context.Context, src *chronograf.Source) error {
+				Logger: log.New(log.DebugLevel),
+				UsersStore: &mocks.UsersStore{
+					UpdateF: func(ctx context.Context, user *chronograf.User) error {
 						return nil
 					},
-					RolesF: func(ctx context.Context) (chronograf.RolesStore, error) {
-						return nil, fmt.Errorf("no roles")
-					},
-					UsersF: func(ctx context.Context) chronograf.UsersStore {
-						return &mocks.UsersStore{
-							UpdateF: func(ctx context.Context, u *chronograf.User) error {
-								return nil
-							},
-							GetF: func(ctx context.Context, name string) (*chronograf.User, error) {
-								return &chronograf.User{
-									Name: "marty",
-								}, nil
-							},
+					GetF: func(ctx context.Context, ID string) (*chronograf.User, error) {
+						switch ID {
+						case "1336":
+							return &chronograf.User{
```

Comment:
In this update test the user information stays the same.  I think you should change the arguments so that you can observe that a change happens.

Reply:
yop